scapy/layers/dot15d4.py

```diff
@@ -0,0 +1,405 @@
+            0:"Implicit", 1:"1oKeyIndex", 2:"4o-KeySource-1oKeyIndex", 3:"8o-KeySource-1oKeyIndex"}
+        ),
+        BitEnumField("sec_sc_seclevel", 0, 3, {0:"None", 1:"MIC-32", 2:"MIC-64", 3:"MIC-128",
+                                               4:"ENC", 5:"ENC-MIC-32", 6:"ENC-MIC-64", 7:"ENC-MIC-128"}),
+        XLEIntField("sec_framecounter", 0x00000000), # 4 octets
+        # Key Identifier (variable length): identifies the key that is used for cryptographic protection
+        # Key Source : length of sec_keyid_keysource varies btwn 0, 4, and 8 bytes depending on sec_sc_keyidmode
+        # 4 octets when sec_sc_keyidmode == 2
+        ConditionalField(XLEIntField("sec_keyid_keysource", 0x00000000), 
+            lambda pkt:pkt.getfieldval("sec_sc_keyidmode") == 2),
+        # 8 octets when sec_sc_keyidmode == 3
+        ConditionalField(LELongField("sec_keyid_keysource", 0x0000000000000000), 
+            lambda pkt:pkt.getfieldval("sec_sc_keyidmode") == 3),
+        # Key Index (1 octet): allows unique identification of different keys with the same originator
+        ConditionalField(XByteField("sec_keyid_keyindex", 0xFF), 
+            lambda pkt:pkt.getfieldval("sec_sc_keyidmode") != 0),
+    ]
+
+class Dot15d4Data(Packet):
+    name = "802.15.4 Data"
+    fields_desc = [
+                    XLEShortField("dest_panid", 0xFFFF),
+                    dot15d4AddressField("dest_addr", 0xFFFF, length_of="fcf_destaddrmode"),
+                    ConditionalField(XLEShortField("src_panid", 0x0), \
+                                        lambda pkt:util_srcpanid_present(pkt)),
+                    ConditionalField(dot15d4AddressField("src_addr", None, length_of="fcf_srcaddrmode"), \
+                                        lambda pkt:pkt.underlayer.getfieldval("fcf_srcaddrmode") != 0),
+                    # Security field present if fcf_security == True
+                    ConditionalField(PacketField("aux_sec_header", Dot15d4AuxSecurityHeader(), Dot15d4AuxSecurityHeader),
+                                        lambda pkt:pkt.underlayer.getfieldval("fcf_security") == True),
+                    ]
+    def mysummary(self):
+        return self.sprintf("802.15.4 Data ( %Dot15d4Data.src_panid%:%Dot15d4Data.src_addr% -> %Dot15d4Data.dest_panid%:%Dot15d4Data.dest_addr% )")
+
+    def guess_payload_class(self, payload):
+        # TODO: find a better algorithm to guess the payload
+        from scapy.layers import zigbee
+        from scapy.layers import sixlowpan
+        ## FIXME: xbee is still on early stage so it is disabled for now
+        # from scapy.layers import xbee
+        # if Padding not in xbee.Xbee(payload):
+        #     return xbee.Xbee
+        if Padding not in zigbee.ZigbeeNWK(payload):
+            return zigbee.ZigbeeNWK
+        if Padding not in sixlowpan.SixLoWPAN(payload):
+            return sixlowpan.SixLoWPAN
+        return Packet.guess_payload_class(self, payload)
```

Comment:
You will need to include the sixlowpan, zigbee and xbee modules (I’ve seen them in the repo you took this code from)